diff -ruN a/c++/Makefile.in b/c++/Makefile.in
--- a/c++/Makefile.in	2025-10-05 04:15:57.000000000 +0800
+++ b/c++/Makefile.in	2025-10-11 21:52:05.454891853 +0800
@@ -119,7 +119,7 @@
 		-l@FORM_NAME@@ABI_SUFFIX@ \
 		-l@MENU_NAME@@ABI_SUFFIX@ \
 		-l@PANEL_NAME@@ABI_SUFFIX@ \
-		-lncurses@ABI_SUFFIX@ @SHLIB_LIST@
+		-lncurses@ABI_SUFFIX@
 
 LIBROOT		= ncurses++
 
@@ -158,8 +158,7 @@
 LDFLAGS_DEFAULT	= $(LINK_@DFT_UPR_MODEL@) $(LDFLAGS_@DFT_UPR_MODEL@)
 
 # flags for library built by this makefile
-LDFLAGS		= $(TEST_ARGS) @LDFLAGS@ \
-	@LD_MODEL@ $(TEST_LIBS) @LIBS@ $(CXXLIBS)
+LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@ $(CXXLIBS)
 
 AUTO_SRC	= \
 		etip.h
diff -ruN a/form/Makefile.in b/form/Makefile.in
--- a/form/Makefile.in	2025-10-05 04:15:57.000000000 +0800
+++ b/form/Makefile.in	2025-10-11 21:52:52.357822081 +0800
@@ -115,7 +115,7 @@
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
diff -ruN a/menu/Makefile.in b/menu/Makefile.in
--- a/menu/Makefile.in	2025-10-05 04:15:57.000000000 +0800
+++ b/menu/Makefile.in	2025-10-11 21:53:25.875057007 +0800
@@ -115,7 +115,7 @@
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
diff -ruN a/panel/Makefile.in b/panel/Makefile.in
--- a/panel/Makefile.in	2025-10-05 04:15:57.000000000 +0800
+++ b/panel/Makefile.in	2025-10-11 21:54:20.731803796 +0800
@@ -117,7 +117,7 @@
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@ABI_SUFFIX@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
